Add an optional stick tilt input filter

MaEtUgR opened the pull request:
**Describe problem solved by this pull request**
An always reoccurring pattern seems people detuning the rate and attitude control loops because "the drone flies too aggressive when moving the sticks".

These are two separate things:
1. The controller trying to follow the setpoint as close as possible and in an ideal world would achieve perfect tracking even with disturbances like wind, motor tolerances and friction, unconsidered aerodynamic effects of the propellers, unbalanced mass distribution, ...
2. Considering perfect tracking of the controller how would the vehicle move given a certain user input. In other words how the setpoint changes when you move the stick.

**Describe your solution**
As discussed with @bresch I used a simple first-order IIR low pass to filter the "x, y" stick input that determines the tilt angle of the setpoint. As we've seen with the yaw speed setpoint filter this results in intuitive flight characteristics that feel like the vehicle dynamics are a bit slower e.g. like a heavier vehicle see https://github.com/PX4/Firmware/blob/8804dae4808ebcb849c02b78b446ea01f4aa45ad/src/lib/flight_tasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp#L101-L104

![image](https://user-images.githubusercontent.com/4668506/81690248-52c81300-945b-11ea-9f3e-8c550b1ae3fe.png)

**Test data / coverage**
Tested in SITL, QAV250 and a bigger vehicle with all kinds of settings. `MC_MAN_TILT_TAU` of 0.15 seems to be a sweet spot between filtering crazy input but keeping high responsiveness to stick input.

**Additional context**
In order to fully resolve https://github.com/PX4/Firmware/issues/12971 I'll also implement a solution for altitude mode.

dagar commented:
> MC_MAN_TILT_TAU of 0.15 seems to be a sweet spot between filtering crazy input but keeping high responsiveness to stick input.

Did you consider setting a conservative default value for the new parameter?

MaEtUgR commented:
> Did you consider setting a conservative default value for the new parameter?

I did but I'm almost certain that experienced pilots who never mentioned "too aggressive" don't expect to have filtered stick input in stabilized. Should I set it to 0.1s time constant or so? The difference can be felt and to generate strong step inputs for tuning it's useful to disable filtering...

dagar commented:
> Should I set it to 0.1s time constant or so? The difference can be felt and to generate strong step inputs for tuning it's useful to disable filtering...

Trying to find good defaults is always a losing battle, and I don't know what's appropriate here. Personally I would try to optimize for the hypothetical "common" case of someone assembling something like an s500 or equivalent airframe and expecting it to mostly work first try. The case where a 30 Hz default `IMU_GYRO_CUTOFF` is appropriate...

Also note that `RC_FLT_CUTOFF` default is 10 Hz.

MaEtUgR commented:
> Also note that `RC_FLT_CUTOFF` default is 10 Hz.

Right, when all modes have their smoothed interpretation of stick input that filter can go away.

bresch commented:
5 bytes overflow on v2...

MaEtUgR commented:
I added the same filter also for altitude mode. We tested that today and it works as expected. Then I removed the 2nd order RC filter.

a) to save some flash
b) I think it's useless (I know it was me adding it in the first place)
c) people don't even know it exists
d) It depended on RC update rate
e) it doesn't apply to joystick input
f) it applies to all flight modes the same way

MaEtUgR commented:
Successfully flight tested in https://logs.px4.io/plot_app?log=67cd2a12-1f04-4bec-affe-2ff3db8b6417
I consider it ready to merge.

bkueng commented:
@MaEtUgR can you please update the airframe configs to remove `RC_FLT_CUTOFF` as well?

MaEtUgR commented:
@bkueng Better later than never: https://github.com/PX4/Firmware/pull/15391
